### PR TITLE
CORE-263 pack hash

### PR DESCRIPTION
**Pull request by rydrman**
support accessing the has of a pack, and fix a couple of bugs that showed up while doing so

### CODE REVIEW

File: libpac/database.go [hunk 1]
```diff
@@ -31,6 +31,9 @@ type databaseRef struct {
 // for tracking in python
 func newDatabaseRef(db storage.Database) (*databaseRef, error) {
 
+	if db == nil {
```
Comment (malkhamis) [inline libpac/database.go]: is `db` a pointer? Will that comparison do what it is intended to do? I see that `storage.Database` is an interface. This also goes for `func newEntryRef(e pac.Entry) (*entryRef, error)`

File: src/pack.c [hunk 2]
```diff
@@ -49,11 +51,16 @@ static PyMethodDef PackMethods[] = {
 	{NULL}
 };
 
+static PyMemberDef PackMembers[] = {
```
Comment (malkhamis) [inline src/pack.c]: wanna `const` that if you don't expect it to change ? :p

Comment (rydrman) [reply src/pack.c]: you're all about the const, eh? `PyTypeObject.tp_members` is a non-const array

File: libpac/database.go (hunk 1 above)
Comment (rydrman) [reply libpac/database.go]: well if it's not a pointer then it won't be nil, so no harm done...

**Pull request merged**